cpp/src/neighbors/mg/snmg.cuh

```diff
@@ -605,6 +605,9 @@ void search(const raft::resources& clique,
                                        n_neighbors,
                                        n_batches);
     } else if (merge_mode == TREE_MERGE) {
+      ASSERT(index.num_ranks_ % 2 == 0,
```

Comment:
We discussed this briefly in our meeting, but I think we should find a better alternative here in the short term. I don't mind merging this change so that users at least have a warning / reason instead of segfaulting, but I don't think having an even number of GPUs is a fair requirement to levvy on our users. At the very least, we should be able to pair off the reductions using P2P, where there may be 1 GPU that only contributes in the final merge stage.

Reply:
Sure, created an issue over here : https://github.com/rapidsai/cuvs/issues/979. Just to be clear, SNMG ANN sharded indexes can still be searched with an odd GPU count with the `MERGE_ON_ROOT_RANK` mode.